iis-wf/iis-wf-export-actionmanager/src/test/java/eu/dnetlib/iis/wf/export/actionmanager/sequencefile/FieldAccessor.java

```diff
@@ -0,0 +1,143 @@
+package eu.dnetlib.iis.wf.export.actionmanager.sequencefile;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Accesses object fields using getter methods of field names encoded in properties file as
+ * keys. Nested field names are separated with dots.
+ * 
+ * Notice: field names should start with capital letters as they appear in getter method.
+ * 
+ * All fields requiring decoder should be marked with '$' prefix. 
+ * 
+ * @author mhorst
+ *
+ */
+public class FieldAccessor {
+
+	private static final char FIELD_PATH_SEPARATOR = '.';
+
+	private static final char DECODER_PREFIX = '$';
+
+	private static final String GETTER = "get";
+
+	private Map<String, FieldDecoder> decoders = new HashMap<String, FieldDecoder>();
+
+	/**
+	 * Registers field decoder.
+	 * 
+	 * @param field
+	 * @param decoder
+	 */
+	public void registerDecoder(String field, FieldDecoder decoder) {
+		decoders.put(field, decoder);
+	}
+
+	/**
+	 * Extracts value from given object based on fieldPath.
+	 * @param fieldPath field names acessible vie getter methods, nested fields are separated with dots
+	 * @param object object value should be extracted from
+	 * @return extracted value
+	 * @throws Exception
+	 */
+	public Object getValue(String fieldPath, Object object) throws Exception {
```

Comment:
I've just applied simple patch allowing subelement extraction from decoded object. Appropriate unit test was written for this case.

We do not support decoding array/list subelement though as it doesn't seem to be required.